ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -232,12 +235,110 @@ public async Task<Response<List<BuyerDTO>>> GetBidBuyers(string bidId)
+            {
+                return new Response<List<BidDTO>>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = validationErorrString };
+            }
+            else // TODO : extract to sub methods
+            {
+                IEnumerable<BidEntity> filteredBids = _context.Bids
+                    // filter by Categories
+                    .Where(bid => bid.Category == null ?
+                                true :
+                                bid.Category.Equals(bidsFilters.Category) &&
+                                (bidsFilters.SubCategory == null) ?
+                                    true :
+                                    bid.SubCategory != null && bid.SubCategory.Equals(bidsFilters.Category))
+                    //filter by prices
+                    .Where(bid => bidsFilters.MaxPrice < Int32.MaxValue ? bid.MaxPrice < bidsFilters.MaxPrice : true &&
+                            bidsFilters.MinPrice > 0 ? bid.MaxPrice > bidsFilters.MinPrice : true)
+                    // filter by query string
+                    .Where(bid => bid.Product.Name.Contains(bidsFilters.Search) || bid.Product.Description.Contains(bidsFilters.Search));
+                
+                //sort
+                IEnumerable<BidEntity> sortedBids;
+                BidsSortByOrder sortOrder = bidsFilters.SortOrder;
+                switch (bidsFilters.SortBy)
+                {
+                    case BidsSortByOptions.CreationDate:
+                        sortedBids = filteredBids.Sort(bid => bid.CreationDate, sortOrder);
+                        break;
+                    case BidsSortByOptions.ExpirationDate:
+                        sortedBids = filteredBids.Sort(bid => bid.ExpirationDate, sortOrder);
+                        break;
+                    case BidsSortByOptions.SupplierProposals:
+                        sortedBids = filteredBids.Sort(bid => bid.PotenialSuplliersCounter, sortOrder);
+                        break;
+                    case BidsSortByOptions.DemandedItems:
+                        sortedBids = filteredBids.Sort(bid => bid.UnitsCounter, sortOrder);
+                        break;
+                    case BidsSortByOptions.Price:
+                        sortedBids = filteredBids.Sort(bid => bid.MaxPrice, sortOrder);
+                        break;
+                    default:
+                        sortedBids = filteredBids.Sort(bid => bid.MaxPrice, sortOrder);
+                        break;
+                }
+
+                // return page
+                int pageSize = bidsFilters.Limit == 0 || bidsFilters.Limit > _maxPageSize ? _pageDefaultSize : bidsFilters.Limit;
+
+                var bidsPage = sortedBids
+                    .Skip(bidsFilters.Page * pageSize)
+                    .Take(pageSize)
+                    .Select (bidEntity => _mapper.Map<BidDTO>(bidEntity))
+                    .ToList();
```

Comment:
you can make this async

Reply:
and maybe the prior queries as well

Reply:
@yotamc4 OrderBy, Skip, Take, Include,Where doesn't have async versions...
I've made the changes for async where I can. please take a look at it and check that I didn't make שטויות